### ISSUE TITLE
🔴 Fix Feature Quality Issues

### ISSUE THREAD

**Issue by gtlntw** [gtlntw/TradingBot-8888#2]
## Problem
Current feature engineering has several critical issues affecting model performance:
- **Missing Data**: Many NaN values in early rows due to technical indicators requiring lookback periods
- **Static Sentiment**: Fear/Greed index stuck at 50.0 (not real market sentiment)
- **Scale Mismatch**: Features have vastly different scales (prices vs ratios vs percentages)

## Root Cause Analysis
1. **NaN Values**: Technical indicators like SMA, RSI need warm-up periods
2. **Fake Sentiment**: Simulated sentiment data instead of real market sentiment
3. **No Scaling**: Raw features fed to models without normalization

### 1. Handle Missing Data
- [ ] **Forward fill strategy**: Use previous values for initial NaN periods
- [ ] **Minimum data requirements**: Skip training until sufficient lookback data available
- [ ] **Alternative indicators**: Use indicators with shorter lookback periods for early data

### 2. Real Sentiment Data Integration
- [ ] **Fear & Greed API**: Integrate real CNN Fear & Greed Index
- [ ] **News sentiment**: Add news headline sentiment analysis
- [ ] **Social media**: Twitter/Reddit sentiment for crypto markets
- [ ] **On-chain metrics**: Whale movements, exchange flows

### 3. Feature Scaling & Engineering
- [ ] **StandardScaler**: Normalize features to zero mean, unit variance
- [ ] **RobustScaler**: Use for features with outliers
- [ ] **Feature selection**: Remove low-variance and correlated features
- [ ] **Multi-timeframe**: Add 1h, 4h data alongside daily data

## Implementation Plan
```python
# 1. Fix NaN handling
def handle_missing_data(df):
    # Forward fill initial NaN values
    df = df.fillna(method='ffill')
    # Drop rows with remaining NaN
    return df.dropna()

# 2. Add real sentiment
def get_fear_greed_index():
    # Call real Fear & Greed API
    response = requests.get('https://api.alternative.me/fng/')
    return response.json()

# 3. Implement scaling
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)
```

## Success Criteria
- [ ] Zero NaN values in final feature matrix
- [ ] Real-time sentiment data (not static 50.0)
- [ ] All features scaled to similar ranges
- [ ] Feature importance analysis completed
- [ ] Model performance improvement of >10%

## Files to Modify
- `trading_bot/data/features.py` (sentiment data, scaling)
- `trading_bot/data/preprocessor.py` (NaN handling)
- `configs/default.yaml` (sentiment API config)

## External Dependencies
- Fear & Greed Index API
- News sentiment API (optional)
- Social media APIs (optional)

## Priority
🔴 **Critical** - Poor data quality directly impacts all model predictions
